dashpay/src/main/kotlin/org/dashj/platform/dashpay/ContactRequests.kt

```diff
@@ -23,32 +25,45 @@ class ContactRequests(val platform: Platform) {
         val contactKey = contactKeyChain.watchingKey
         val contactPub = contactKey.serializeContactPub()
 
-        val (encryptedContactPubKey, encryptedAccountLabel) = fromUser.encryptExtendedPublicKey(
-            contactPub,
-            toUser,
-            0,
-            aesKey
-        )
-        val accountReference = fromUser.getAccountReference(aesKey, toUser)
-
-        val contactRequestDocument = ContactRequest.builder(platform)
-            .to(toUser.id)
-            .from(fromUser.uniqueIdentifier)
-            .encryptedPubKey(encryptedContactPubKey, toUser.publicKeys[0].id, fromUser.identity!!.publicKeys[0].id)
-            .accountReference(accountReference)
-            .encryptedAccountLabel(encryptedAccountLabel)
-            .build().document
-
-        val transitionMap = hashMapOf(
-            "create" to listOf(contactRequestDocument)
-        )
-
-        val transition = platform.dpp.document.createStateTransition(transitionMap)
-        fromUser.signStateTransition(transition, aesKey)
-
-        platform.broadcastStateTransition(transition)
+        val toUserPublicKey = toUser.publicKeys.find { publicKey ->
+            // is the publicKey disabled?
+            if (publicKey.disabledAt == null || publicKey.disabledAt!! > Date().time) {
+                // is the public key of type ECDSA with a high enough security level?
+                publicKey.type == IdentityPublicKey.Type.ECDSA_SECP256K1 && publicKey.securityLevel <= IdentityPublicKey.SecurityLevel.MEDIUM
+            } else {
+                false
+            }
+        }
```

Comment:
in the previous version, we were just choosing the first key of the other identity.  Now we will choose the first key of the correct type and security level.